### PR TITLE
Give more permission to the diagnose pods

### PR DESCRIPTION
**Pull request by aswinsuryan**
Some containers which run os like rhel/fedora run tcpdump
as specific user id "72". So it needs pods to be privileged.
Also setting the runAsUser prevents the pods from starting with
random user id

* Make the diagnose pod a privileged
* Run the diagnose pod with userId 0

Signed-off-by: Aswin Suryanarayanan <aswinsuryan@gmail.com>

### CODE REVIEW

**Comment by tpantelis** [comment]
Can you explain in the summary why this is needed? Also perhaps a comment in the code.

**Comment by tpantelis** [comment]
> Can you explain in the summary why this is needed? Also perhaps a comment in the code.

OK so this is to fix the `tcpdump` permission issue.

**Comment by dfarrell07** [comment]
> Can you explain in the summary why this is needed? Also perhaps a comment in the code.

+1, let's try to be clear in the commit message and code about what fails exactly without this

**Comment by aswinsuryan** [comment]
> > Can you explain in the summary why this is needed? Also perhaps a comment in the code.
> 
> +1, let's try to be clear in the commit message and code about what fails exactly without this

Done.

**Review by tpantelis — APPROVED** [review]

File: internal/pods/schedule.go [hunk 1]
```diff
@@ -148,12 +148,21 @@ func (np *Scheduled) schedule() error {
 		},
 	}
 
+	// Some containers which run os like rhel/fedora runs tcpdump
+	// as specific user id "72". So it needs pods to be privileged.
+	// Also setting the runAsUser prevent the pods from starting with
+	// random user id
+	privileged := true
+	runAsUser := int64(0)
+
```
Comment (sridhargaddam) [inline internal/pods/schedule.go]: Okay, lets move this to Line159

Just for info, currently, we use privilege pods for two use-cases.
1. For identifying the hostname when it does not match with the name in the endpoint
2. As part of diagnose firewall validation 

In the other cases, when this API is called with podNetworking, we use commands like "curl" or "nc".

**Review by sridhargaddam — CHANGES_REQUESTED** [review]

File: internal/pods/schedule.go [hunk 2]
```diff
@@ -148,12 +148,21 @@ func (np *Scheduled) schedule() error {
...
 	if np.Config.Scheduling.Networking == HostNetworking {
 		networkPod.Spec.Containers[0].SecurityContext = &v1.SecurityContext{
 			Capabilities: &v1.Capabilities{
 				Add:  []v1.Capability{"NET_ADMIN", "NET_RAW"},
 				Drop: []v1.Capability{"all"},
 			},
+			Privileged: &privileged,
+			RunAsUser:  &runAsUser,
```
Comment (skitt) [inline internal/pods/schedule.go]: Please use `k8s.io/utils/pointer`:

```suggestion
			Privileged: pointer.Bool(true),
			RunAsUser:  pointer.Int64(0),
```

**Review by skitt — CHANGES_REQUESTED** [review]

**Review by skitt — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Comment by sridhargaddam** [comment]
@aswinsuryan I can handle the backport.

**Comment by sridhargaddam** [comment]
Backport PR: https://github.com/submariner-io/subctl/pull/435

**Pull request merged**